monkey/tests/unit_tests/monkey_island/cc/services/authentication_service/token/test_token_validator.py

```diff
@@ -50,3 +50,44 @@ def test_validate_refresh_token__invalid(freezer):
 
     with pytest.raises(BadData):
         token_validator.validate_token(invalid_token)
+
+
+def test_get_token_payload():
+    token_expiration_timedelta = 1 * 60  # 1 minute
+    payload = "fake_user_id"
+
+    app, _ = build_app()
+    token_generator = TokenGenerator(app.security)
+    token_validator = TokenValidator(app.security, token_expiration_timedelta)
+
+    token = token_generator.generate_token(payload)
+
+    assert token_validator.get_token_payload(token) == payload
+
+
+def test_get_token_payload__expired_token(freezer):
+    token_expiration = 1 * 60  # 1 minute
+    generation_time = "2020-01-01 00:00:00"
+    validation_time = "2020-01-01 00:03:30"
+    payload = "fake_user_id"
+
+    app, _ = build_app()
+    token_generator = TokenGenerator(app.security)
+    freezer.move_to(generation_time)
+    token = token_generator.generate_token(payload)
+    token_validator = TokenValidator(app.security, token_expiration)
+    freezer.move_to(validation_time)
+
+    with pytest.raises(SignatureExpired):
+        token_validator.get_token_payload(token)
+
+
+def test_get_token_payload__invalid_token():
+    token_expiration = 1 * 60  # 1 minute
+    invalid_token = "invalid_token"
+
+    app, _ = build_app()
+    token_validator = TokenValidator(app.security, token_expiration)
+
+    with pytest.raises(BadData):
+        token_validator.get_token_payload(invalid_token)
```

Comment:
I'm not sure we need these tests. Validating the token is done by `validate_token`, I don't think we need to assert this behavior for decoding the payload. Alternatively, we could just have one method to decode the payload and use that for validation as well, but these are different tasks and might change in the future

Reply:
I think it's worth testing that an error is raised if a bad token is passed to `get_token_payload()`.

Reply:
+1, worth the test. But this will probably change anyway if we [change it to `TokenParser`.](https://github.com/guardicore/monkey/pull/3181#discussion_r1154679927)